src/ensembl/src/shared/components/upload/Upload.test.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import faker from 'faker';
+import times from 'lodash/times';
+import noop from 'lodash/noop';
+import Upload from './Upload';
+
+const onChange = jest.fn();
+
+const defaultProps = {
+  onChange
+};
+
+const event = { preventDefault: noop, stopPropagation: noop };
+
+const renderUpload = (props: any = defaultProps) =>
+  mount(<Upload {...props} />);
+
+const fileContents = faker.random.words();
+
+const generateFile = () => {
+  return new Blob([fileContents], { type: 'text/plain' });
+};
+
+const files = times(faker.random.number(10), () => generateFile());
+const readAsText = jest.fn();
+const addEventListener = jest.fn((_, evtHandler) => {
+  evtHandler();
+});
+const dummyFileReader = { addEventListener, readAsText, result: fileContents };
+
+describe('Upload', () => {
+  let wrapper: any;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    window.FileReader = jest.fn(() => dummyFileReader) as any;
```

Comment:
Can we use the window-service as a proxy for window so that we don't have to mock anything on the window object itself?

Reply:
I can only understand a part of it. Can you please point me to an example?

Reply:
https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/services/storage-service.ts#L37-L38

Reply:
I've made some changes to use the window-service but can you check if that's right, please?